test/middleware_test.rb

```diff
@@ -90,6 +90,42 @@ def test_amzn_trace_id_in_wrong_format
     assert_invalid_headers_response
   end
 
+  def test_redis_5_call_patching
+    RedisValidationMiddleware.reset!
+    configure_middleware
+
+    redis_config = RedisClient.config(host: "127.0.0.1", port: 10)
+    redis = redis_config.new_pool(timeout: 0.5, size: 1)
+    PrometheusExporter::Instrumentation::MethodProfiler.start
+    redis.call("PING") # => "PONG"
+    redis.call("PING") # => "PONG"
+    results = PrometheusExporter::Instrumentation::MethodProfiler.stop
+    # redis client injects a HELLO preamble on reconnection, so we will see more than 2
+    assert(results[:redis][:calls] >= 2, "expecting at least 2 calls")
```

Comment:
I think this assertion is dangerous because it doesn't help us to catch regressions where we start to over increment method calls. For example, we may accidentally be increment twice instead of once and this test will still pass. 

On another note, I tried to add `redis.call("PING")` before `PrometheusExporter::Instrumentation::MethodProfiler.start` and the results of the call was still tracked in results even though method profiling should not have been started at the point in time. I think there is some state being leaked here.

Reply:
I think @byroot may need to provide an interface to allow us to tell redis-client ... hey ... no preamble please. Otherwise there are hello calls and no idea at all how many of them the implementation will make.

Will see if I can teach method profiler to behave a bit better, I have an idea for an improved test

Reply:
I am not sure there is anything I can do here, I tried a few things, unless we figure out a way of cleaning junking the HELLO calls.

Maybe I can register a HELLO remover middleware, so it does not noisy up the testing... will wait on byroot for guidance

Reply:
If you use RESP2 protocol, there's no `HELLO` and that's what the `redis-rb` gem uses anyway: `RedisClient.new(protocol: 2)`